src/backend/distributed/utils/colocation_utils.c

```diff
@@ -0,0 +1,189 @@
+/*-------------------------------------------------------------------------
+ *
+ * colocation_utils.c
+ *
+ * This file contains functions to perform useful operations on co-located tables.
+ *
+ * Copyright (c) 2014-2016, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#include "postgres.h"
+
+#include "access/genam.h"
+#include "access/heapam.h"
+#include "access/htup_details.h"
+#include "distributed/colocation_utils.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/multi_logical_planner.h"
+#include "utils/rel.h"
+
+
+static List * DistTableOidList(void);
+
+
+/*
+ * TableColocationId function returns co-location id of given table. This function errors
+ * out if given table is not distributed.
+ */
+int
+TableColocationId(Oid distributedTableId)
+{
+	DistTableCacheEntry *cacheEntry = DistributedTableCacheEntry(distributedTableId);
+
+	return cacheEntry->colocationId;
+}
+
+
+/*
+ * TablesColocated function checks whether given two tables are co-located and
+ * returns true if they are co-located. This function errors
+ * out if given tables is not distributed.
+ */
+bool
+TablesColocated(Oid leftDistributedTableId, Oid rightDistributedTableId)
+{
+	int leftClocationId = TableColocationId(leftDistributedTableId);
+	int rightColocationId = TableColocationId(rightDistributedTableId);
+
+	if (leftClocationId == INVALID_COLOCATION_ID ||
+		rightColocationId == INVALID_COLOCATION_ID)
+	{
+		return false;
+	}
+
+	return leftClocationId == rightColocationId;
+}
+
+
+/*
+ * ShardsColocated function checks whether given two shards are co-located and
```

Comment:
I think this comment is missing the information that we also consider the tables of the given shards are co-located